src/DebugEngineHost/HostNatvisProject.cs

```diff
@@ -89,6 +102,20 @@ public static void FindNatvisInSolutionImpl(List<string> paths)
                 }
             }
 
+            public static string FindSolutionRootImpl()
+            {
+                string root = null;
+                string slnFile;
+                string slnUserFile;
+                var solution = (IVsSolution)Package.GetGlobalService(typeof(SVsSolution));
+                if (solution == null)
+                {
+                    return null; // failed to find a solution
+                }
+                solution.GetSolutionInfo(out root, out slnFile, out slnUserFile);
+                return root;
```

Comment:
So getting the "solution" does work in the open folder case, however it was not being invoked in the attach code path. Fixed.